src/packager.ts

```diff
@@ -79,14 +92,92 @@ export const handler: ScheduledHandler = Sentry.wrapHandler(
 const concatAllObjects = async (
   client: S3Client,
   bucket: string,
-  prefix: string,
+  sourcePrefix: string,
+  outputPrefix: string,
+  serviceDay: string,
+  recoveryPrefix: string,
   filename: string
 ) => {
+  const prefix = path.posix.join(sourcePrefix, serviceDay);
   const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "temp-"));
   const outputPath = path.join(tempDir, filename);
   await fs.mkdir(path.dirname(outputPath), { recursive: true });
   const outputFile = await fs.open(outputPath, "w");
 
+  if (recoveryPrefix) {
+    for await (const {
+      $metadata: metadata,
+      Contents: objects,
+    } of paginateListObjectsV2(
+      { client },
+      { Bucket: bucket, Prefix: recoveryPrefix }
+    )) {
+      if (objects === undefined)
+        throw exception("BadS3Response", JSON.stringify(metadata));
+
+      for (const { Key: key } of objects) {
+        const command = new GetObjectCommand({ Bucket: bucket, Key: key });
+        const { $metadata: metadata, Body: data } = await client.send(command);
+
+        if (data === undefined)
+          throw exception("BadS3Response", JSON.stringify(metadata));
+
+        if (
+          key &&
+          key.startsWith(`ocs-saver-${outputPrefix}-1-${serviceDay}`)
```

Comment:
question: this looks like it's the same for all files being recovered. could you include this as a part of the `Prefix` passed to AWS and limit the number of files that need to be scanned?